### PR TITLE
Add minimum teaching days

### PR DESCRIPTION
**Pull request by rhnonose**

### CODE REVIEW

**Comment by ghatighorias** [comment]
What about

**Comment by ghatighorias** [comment]
Sorry for the Mistake :P

File: web/models/terms/term.ex [hunk 1]
```diff
@@ -21,11 +22,29 @@ defmodule CoursePlanner.Terms.Term do
...
-    |> validate_required([:name, :start_date, :end_date])
+    |> cast(params, [:name, :start_date, :end_date, :minimum_teaching_days])
+    |> validate_required([:name, :start_date, :end_date, :minimum_teaching_days])
     |> validate_date_range()
   end
 
+  def validate_minimum_teaching_days(%{valid?: true} = changeset, holidays) do
+    teaching_days = count_teaching_days(changeset)
+    min = Changeset.get_field(changeset, :minimum_teaching_days)
+    if teaching_days - length(holidays) > min do
```
Comment (ggpasqualino) [inline web/models/terms/term.ex]: what do you think of moving `teaching_days - length(holidays)` to the function `count_teaching_days`?

File: test/models/term_test.exs [hunk 2]
```diff
@@ -36,9 +37,39 @@ defmodule CoursePlanner.TermTest do
...
+      "minimum_teaching_days" => 3,
+      "holidays" => %{
+        "0" => %{name: "Labor Day 1", date: %{day: 18, month: 3, year: 2017}},
+        "1" => %{name: "Labor Day 2", date: %{day: 19, month: 3, year: 2017}}
+      }
+    })
+    assert length(term.holidays) == 2
+  end
+
+  test "term doesn't have minimum teaching days" do
```
Comment (ggpasqualino) [inline test/models/term_test.exs]: could you add a test which has no holidays and the term doesn't have enough teaching days anyways?

Comment (rhnonose) [inline test/models/term_test.exs]: Yeah, forgot that case.

**Pull request merged**